### ISSUE TITLE
Deleting cascade single record

### ISSUE THREAD

**Issue by Joseph82** [awslabs/aws-mobile-appsync-sdk-android#225]
Suppose you have a query `getUsers()` which gives back 100 items, I will expect in the AppSyncStore the data to be stored in 102 records: 
1 record for the `ROOT`
1 record for the query. Something like `getUsers -> {"__typename":"UserResult","users":[...]}`
100 records, one for each pair `key - object (my User object)`

Now if I want to delete all of them, it seems to me that there is no other way than going through the list of items (100) and delete one by one. 

I was wondering if there is a way for deleting in cascade the 100 records, with something like:

`awsAppSyncClient.getStore().remove("my_query_key")`

so, basically deleting only the `getUsers` query.

**Comment by kvasukib** [comment]
@Joseph82 Thank you for reporting to us. Have you tried using the `awsAppSyncClient.getStore().remove(CacheKey)` method? 

See https://github.com/awslabs/aws-mobile-appsync-sdk-android/blob/master/aws-android-sdk-appsync/src/main/java/com/amazonaws/mobileconnectors/appsync/cache/normalized/AppSyncStore.java#L65 for reference.

**Comment by Joseph82** [comment]
@kvasukib thanks for the answer. Yes, as I mention I am aware of the existence of `AppSyncStore.remove()` method, but it seems like all keys are independent. So if I remove a key whose value contains a reference to other keys, those keys are not gonna be deleted.

I actually already implemented it manually in this way:

```fun AWSAppSyncClient.clearAllKeysFor(operationName: String, responseName: String) {
    val map = this.store.read(CacheKey.from("QUERY_ROOT").key(), CacheHeaders.NONE)
        .fields()
    map?.filter {
        it.key.startsWith(operationName)
    }?.forEach { cacheItem ->
        val fields = this.store.read(CacheKey.from(cacheItem.key).key(), CacheHeaders.NONE).fields()
        (fields[responseName] as List<CacheReference>).forEach { item ->
            this.store.remove(CacheKey.from(item.key())).execute()
        }
        this.store.remove(CacheKey.from(cacheItem.key)).execute()
    }
}```

But I am not sure this is the best solution: deleting something by accident can have deep consequences on the whole stability of the app.

So, I was wondering if there is a way for automatically remove all key-value pairs that are related to a specific query.
